packages/core/src/pagination/pagination-button.element.ts

```diff
@@ -45,58 +45,46 @@ import styles from './pagination-button.element.scss';
  * @cssprop --box-shadow
  * @cssprop --min-height
  * @cssprop --min-width
+ * @property disabled
  */
 
-export class CdsPaginationButton extends CdsBaseButton {
+export class CdsPaginationButton extends CdsAction {
   /**
    * Sets the action from a predefined list of actions
    */
-  @property({ type: String })
-  action: 'first' | 'prev' | 'next' | 'last';
-
-  connectedCallback() {
-    super.connectedCallback();
-    this.type = 'button';
-  }
-
-  @querySlot('cds-icon', { assign: 'cds-icon-slot' }) cdsIcon: HTMLElement;
+  @property({ type: String }) action: 'first' | 'prev' | 'next' | 'last';
 
-  @query('cds-icon') private cdsIconDefault: HTMLElement;
+  @state() private direction: 'up' | 'right' | 'down' | 'left';
 
-  private get customContent() {
-    return !this.action && !this.cdsIconDefault;
+  get cdsIcon() {
```

Comment:
Why wouldn't a `@query()` property work here?

Reply:
Updated to use query